### ISSUE TITLE
ViewChangeEvent.getParameterMap() should include query parameters

### ISSUE THREAD

**Issue by mvysny** [vaadin/framework#10596]
Vaadin 8.3. Since the addition of HTML5 History API the fragments has been abolished and REST-like paths are now being used. The query parameters were generally not used with the old fragment approach (since they are required to be present before fragment by the URL spec and that made the URLs look weird), but now they're back in the game.

However, the `ViewChangeEvent.getParameterMap()` still only returns the parameter map from the first path parameter. For example:

* `http://localhost:8080/myview/param1=value1` makes parameterMap return a map `{param1=value1}`; however
* `http://localhost:8080/myview?param1=value1` makes parameterMap return an empty map

I would suggest the parameterMap to actually return a union of path parameters and query parameters.

**Comment by elmot** [comment]
I do not think we can do that. This is behavior change, existing applications may be affected with that. Query parameters are still available with 
`ServletRequest.getParameter()` or `PortletRequest.getParameter()`, and that is not a big deal to add a couple of lines to the application code for getting parameters from both places. If you need this feature in many applications, feel free to create an addon.
